src/transforms/transform-deployment.ts

```diff
@@ -48,30 +53,89 @@ const getLastSuccessfulDeployCommitSha = async (
+
+		statsd.increment(metricDeploymentPersistent.lookup, tags, info);
+
+		const lastSuccessful = await findLastSuccessDeployment({
+			gitHubBaseUrl: githubInstallationClient.baseUrl,
+			env: currentDeployEnv,
+			repositoryId: repoId,
+			currentDate: new Date(currentDeployDate)
+		}, logger);
+
+		if (!lastSuccessful) {
+			logger.info("Couldn't find last success deployment from dynamodb");
+			statsd.increment(metricDeploymentPersistent.miss, { missedType: "not-found", ...tags }, info);
+			return undefined;
+		} else if (!lastSuccessful.commitSha) {
+			logger.warn("Missing commit sha from deployment");
+			statsd.increment(metricDeploymentPersistent.miss, { missedType: "sha-empty", ...tags }, info);
+			return undefined;
+		} else {
+			logger.info("Found last success deployment info");
+			statsd.increment(metricDeploymentPersistent.hit, tags, info);
+			return lastSuccessful.commitSha;
+		}
+
+	} catch (e) {
+		statsd.increment(metricDeploymentPersistent.failed, { failType: "lookup", ...tags }, info);
+		logger.error({ err: e }, "Error look up deployment information from dynamodb");
+		throw e;
+	}
+};
+
 const getCommitsSinceLastSuccessfulDeployment = async (
+	jiraHost: string,
+	type: "backfill" | "webhook",
 	owner: string,
+	repoId: number,
 	repoName: string,
 	currentDeploySha: string,
 	currentDeployId: number,
 	currentDeployEnv: string,
+	currentDeployDate: string,
 	githubInstallationClient: GitHubInstallationClient,
 	logger: Logger
 ): Promise<CommitSummary[] | undefined> => {
 
-	// Grab the last 10 deployments for this repo
-	const deployments: Octokit.Response<Octokit.ReposListDeploymentsResponse> | AxiosResponse<Octokit.ReposListDeploymentsResponse> =
-		await githubInstallationClient.listDeployments(owner, repoName, currentDeployEnv, 10);
-
-	// Filter per current environment and exclude itself
-	const filteredDeployments = deployments.data
-		.filter(deployment => deployment.id !== currentDeployId);
+	let lastSuccessfullyDeployedCommit: string | undefined = undefined;
 
-	// If this is the very first successful deployment ever, return nothing because we won't have any commit sha to compare with the current one.
-	if (!filteredDeployments.length) {
-		return undefined;
+	if (type === "webhook" && await booleanFlag(BooleanFlags.USE_DYNAMODB_FOR_DEPLOYMENT_WEBHOOK, jiraHost)) {
+		lastSuccessfullyDeployedCommit = await getLastSuccessDeploymentShaFromDyanmoDB(type, jiraHost, repoId, currentDeployEnv, currentDeployDate, githubInstallationClient, logger);
 	}
 
-	const lastSuccessfullyDeployedCommit = await getLastSuccessfulDeployCommitSha(owner, repoName, githubInstallationClient, filteredDeployments, logger);
+	if (!lastSuccessfullyDeployedCommit) {
+		// Grab the last 10 deployments for this repo
+		const deployments: Octokit.Response<Octokit.ReposListDeploymentsResponse> | AxiosResponse<Octokit.ReposListDeploymentsResponse> =
+			await githubInstallationClient.listDeployments(owner, repoName, currentDeployEnv, 10);
```

Comment:
I was thinking about this... If we treat the Db as a cache, and if the last deployment is not there for some reason (e.g. we dropped the cache) and there was a latency in processing, which makes first 10 items irrelevant... should we fix this logic a little bit?

 - if the last successful deployment is not cached
 - if the CURRENT deployment is not among last 10 deployments OR the last successful deployment is not in the last 10 items
 --- then just use issue key from commit message/branch name (<--- new logic)
 --- otherwise proceed with the current logic
 
wdyt?

Reply:
I am thinking another PR, another day ;ol.
`--- then just use issue key from commit message/branch name (<--- new logic)` this can be unexpected, we can chat more in detail. Basically calling github api `/compare/deployment_sha...<whatever_sha_from_anywhere>` will yield much more info than to simply `just use issue key from commit message`. 
We already calling that api right now for whatever reason, changing this will need more thinking. I suspect previously I compare the deployment only to futures and skip the sha compare api coz some customer tickets.